### PR TITLE
[v7r2] Introduce DIRAC_GFAL_GRIDFTP_ENABLE_IPV6 environment variable

### PR DESCRIPTION
**Pull request by chaen**
Addresses  https://github.com/DIRACGrid/DIRAC/issues/5401 and https://github.com/DIRACGrid/DIRAC/issues/5393

BEGINRELEASENOTES

*Resources
NEW: DIRAC_GFAL_GRIDFTP_ENABLE_IPV6 to control gridftp behavior

ENDRELEASENOTES

### CODE REVIEW

**Comment by chaen** [comment]
@iueda can you please check ?

**Review by chrisburr — APPROVED** [review]

File: docs/source/AdministratorGuide/ServerInstallations/environment_variable_configuration.rst [hunk 1]
```diff
@@ -24,8 +24,12 @@ DIRAC_DEPRECATED_FAIL
 DIRAC_FEWER_CFG_LOCKS
   If ``true`` or ``yes`` or ``on`` or ``1`` or ``y`` or ``t``, DIRAC will reduce the number of locks used when accessing the CS for better performance (default, ``no``).
 
+DIRAC_GFAL_GRIDFTP_ENABLE_IPV6
+  If set to ``false`` or ``no``, disable IPv6 for the GRIDFT plugin (default true).
```
Comment (andresailer) [inline docs/source/AdministratorGuide/ServerInstallations/environment_variable_configuration.rst:28]:
```suggestion
  If set to ``false`` or ``no``, disable IPv6 for the GRIDFTP plugin (default true).
```

File: src/DIRAC/Resources/Storage/GFAL2_StorageBase.py [hunk 2]
```diff
@@ -14,6 +14,8 @@
 * DIRAC_GFAL_GRIDFTP_SESSION_REUSE: This should be exported
   and set to true in server bashrc files for efficiency reasons.
 
+* DIRAC_GFAL_GRIDFTP_ENABLE_IPV6: this should be eported and set
```
Comment (andresailer) [inline src/DIRAC/Resources/Storage/GFAL2_StorageBase.py:17]:
```suggestion
* DIRAC_GFAL_GRIDFTP_ENABLE_IPV6: this should be exported and set
```

Comment (andresailer) [reply src/DIRAC/Resources/Storage/GFAL2_StorageBase.py:17]: For alphabetic order, this variable should come first

File: docs/source/AdministratorGuide/ServerInstallations/environment_variable_configuration.rst [hunk 3]
```diff
@@ -24,8 +24,12 @@ DIRAC_DEPRECATED_FAIL
...
+  See the information in the :ref:`resourcesStorageElement` page.
+
 DIRAC_GFAL_GRIDFTP_SESSION_REUSE
-  If set to ``true`` or ``yes`` the GRIDFT SESSION RESUSE option will be set to True, should be set on server
+  If set to ``true`` or ``yes`` the GRIDFT SESSION REUSE option will be set to True, should be set on server
```
Comment (andresailer) [inline docs/source/AdministratorGuide/ServerInstallations/environment_variable_configuration.rst:32]:
```suggestion
  If set to ``true`` or ``yes`` the GRIDFTP SESSION REUSE option will be set to True, should be set on server
```

File: docs/source/AdministratorGuide/ServerInstallations/environment_variable_configuration.rst (hunk 1 above)
Comment (chaen) [reply docs/source/AdministratorGuide/ServerInstallations/environment_variable_configuration.rst:28]: Done

File: src/DIRAC/Resources/Storage/GFAL2_StorageBase.py (hunk 2 above)
Comment (chaen) [reply src/DIRAC/Resources/Storage/GFAL2_StorageBase.py:17]: done

File: docs/source/AdministratorGuide/ServerInstallations/environment_variable_configuration.rst (hunk 3 above)
Comment (chaen) [reply docs/source/AdministratorGuide/ServerInstallations/environment_variable_configuration.rst:32]: done

**Review by andresailer — APPROVED** [review]

**Review by fstagni — APPROVED** [review]

**Comment by iueda** [comment]
With a patch to `DIRAC/Resources/Storage/GFAL2_StorageBase.py` following this #5402, I confirmed it works on a ipv4-only node.
* Without the patch:
```
dirac-dms-get-file -d /belle/Data/...
...
Trying to download srm://dpm1.egee.cesnet.cz:8446/srm/managerv2?SFN=/dpm/cesnet.cz/home/belle/DATA/belle/Data/...
Failure in plugin to perform getFile Plugin: GFAL2_SRM2 lfn: /belle/Data/..., [70] globus_xio: Unable to connect to 2001:718:19:1::125:24718 globus_xio: globus_xio_system_socket_register_connect failed. globus_xio: System error in connect: Network is unreachable globus_xio: A system call failed: Network is unreachable )
...
```
* With the patch, but without the env.var.:
```
dirac-dms-get-file -d /belle/Data/...
...
Trying to download srm://dpm1.egee.cesnet.cz:8446/srm/managerv2?SFN=/dpm/cesnet.cz/home/belle/DATA/belle/Data/...
Failure in plugin to perform getFile Plugin: GFAL2_SRM2 lfn: /belle/Data/..., [70] globus_xio: Unable to connect to 2001:718:19:1::125:24718 globus_xio: globus_xio_system_socket_register_connect failed. globus_xio: System error in connect: Network is unreachable globus_xio: A system call failed: Network is unreachable )
...
```
* With the env.var.:
```
export DIRAC_GFAL_GRIDFTP_ENABLE_IPV6=false
dirac-dms-get-file -d /belle/Data/...
...
Trying to download srm://dpm1.egee.cesnet.cz:8446/srm/managerv2?SFN=/dpm/cesnet.cz/home/belle/DATA/belle/Data/...
{'Failed': {},
 'Successful': {'/belle/Data/...': '/ext/home/ueda/...'}}

```

**Review by iueda — APPROVED** [review]

**Review by fstagni — APPROVED** [review]

**Pull request merged**